internal/fs/tarfs/tarfs_test.go

```diff
@@ -268,3 +268,25 @@ func TestTarFS_Stat_Unsupported(t *testing.T) {
 		}
 	}
 }
+
+func TestTarFS_New_NonTarFile(t *testing.T) {
+
+	emptyfile := "testdata/emptyfile.tar"
+	nontar := "testdata/nontar.tar"
+	tarwithoutextension := "testdata/tarwithoutextension"
+
+	_, err := New(emptyfile)
+	if want := errdef.ErrInvalidTarFile; !errors.Is(err, want) {
+		t.Errorf("TarFS.New(%s) error = %v, wantErr %v", emptyfile, err, want)
+	}
+
+	_, err = New(nontar)
+	if want := errdef.ErrInvalidTarFile; !errors.Is(err, want) {
+		t.Errorf("TarFS.New(%s) error = %v, wantErr %v", nontar, err, want)
+	}
+
+	_, err = New(tarwithoutextension)
+	if err != nil {
+		t.Errorf("TarFS.New(%s) error = %v, wantErr %v", tarwithoutextension, err, nil)
```

Comment:
nit: The test code can be simplified. You may refer to other tests in this file.

Reply:
Will condense the tests similar to the others.